src/Agent/NewRelic/Agent/Core/Transactions/Transaction.cs

```diff
@@ -249,6 +257,32 @@ public AbstractSegmentData CreateMessageBrokerSegmentData(MessageBrokerDestinati
             return new MessageBrokerSegmentData(brokerVendorName, destinationName, destType, action);
         }
 
+        /// <summary>
+        /// This creates a Datastore segment based on data gathered using the built-in StackExchange.Redis profiling system.
+        /// </summary>
+        /// <param name="invocationTargetHashCode"></param>
+        /// <param name="parsedSqlStatement"></param>
+        /// <param name="connectionInfo"></param>
+        /// <param name="relativeStartTime"></param>
+        /// <param name="relativeEndTime"></param>
+        /// <returns></returns>
+        public ISegment StartStackExchangeRedisSegment(int invocationTargetHashCode, ParsedSqlStatement parsedSqlStatement, ConnectionInfo connectionInfo, TimeSpan relativeStartTime, TimeSpan relativeEndTime)
```

Comment:
This also seems like we are missing an abstraction that prevents leaking some library specific information into the API directly.